HomeWork/HomeWork/Selenium WD/Tests/BookmarksTest.cs

```diff
@@ -1,32 +1,75 @@
+using System.Threading;
 
 namespace HomeWork
 {
     internal class BookmarksTest
     {
         private IWebDriver driver;
-        private UserService service;
-        
+        private ProductCategoryNavigation category;
+        private CategoryPage categoryPage;
+        private CheckboxRuntimeVariable checkboxRuntimeVariables = new CheckboxRuntimeVariable();
+        private ProductPages productPages;
+
         [SetUp]
         public void Setup()
         {
-            driver = new OpenQA.Selenium.Chrome.ChromeDriver();
+            driver = BrowserFactory.CreateDriver();
             driver.Navigate().GoToUrl("https://ek.ua/");
             driver.Manage().Window.Maximize();
-            service = new UserService(driver);
+            category = new ProductCategoryNavigation(driver);
+            categoryPage = new CategoryPage(driver, checkboxRuntimeVariables);
+            productPages = new ProductPages(driver);
+            PageFactory.InitElements(driver, productPages);
         }
 
         [Test]
-        public void Test1()
+        public void TestBookmarks()
         {
-            service.EntryIntoCategoryByName("Гаджеты", "Мобильные");
-            service.SearchBrandsByFilter("Apple");
-            service.AddToBookmarks();
+            Actions actions = new Actions(driver);
+            
+            category.EntryIntoCategoryByName("Гаджеты", "Мобильные");
+
+            categoryPage.SearchBrandByFilter("Apple");
+            categoryPage.VerifyThatCheckboxIsSelected("Apple");
+            categoryPage.ClickOnShowFilterButton();
+
+            Thread.Sleep(1000);
+
+            var tagName = productPages.AddedProductInList.TagName;
+            productPages.SelectProductOnPage("Apple iPhone 13").Click();
+
+            var addedContantButton = driver.FindElement(By.XPath(".//div[starts-with(@class, 'i15-add-content i15-img h')]"));
+            var locationButton = addedContantButton.Location.X;
+            var sizeButton = addedContantButton.Size.Width + locationButton; 
+            actions.MoveToElement(addedContantButton).Perform();
+            var sizeAfterMove = addedContantButton.Size;
+            Assert.AreNotEqual(sizeAfterMove, sizeButton, "Button added content not work");
+            
+            var nameTitleProduct = productPages.FooterWithNameOnPage.Text;
+            productPages.AddedProductInList.Click();
+
+            Thread.Sleep(2000);
+
+            productPages.OpenListWithProduct.Click();
+
+            Thread.Sleep(2000);
+
+            var nameProductInList = productPages.NameProductInSaveList.Text;
+            var textProductInList = nameProductInList.Replace("GB", string.Empty);
+
+            Assert.IsTrue(nameTitleProduct.Contains(textProductInList), "The item added to the bookmark does not match the item in the bookmark");
+            Assert.AreEqual(tagName, "span");
         }
 
         [TearDown]
-        public void Test2()
+        public void Completion()
```

Comment:
maybe better 'AfterTest', because there is described logic that performing after all test business logic?